DuckDuckGo/UnifiedFeedbackForm/MetadataCollectors/UnifiedMetadataCollector.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  UnifiedMetadataCollector.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+protocol UnifiedMetadataCollector {
+    associatedtype Metadata: UnifiedFeedbackMetadata
+
+    func collectMetadata() async -> Metadata?
```

Comment:
Ah, now I see - in a previous comment I mentioned that it doesn't look like we're returning an optional for the metadata, but indeed we are.

Reply:
Good point, this doesn't need to be an optional.